attributes-loader/src/main/scala/it/pagopa/interop/attributesloader/Jobs.scala

```
@@ -0,0 +1,134 @@
+        .filterNot(c => kindToBeExcluded.contains(c.kind)) // Including only Pubbliche Amministrazioni
+        .map(c =>
+          AttributeSeed(
+            code = Option(Digester.toSha256(c.kind.getBytes)),
+            kind = AttributeKind.CERTIFIED,
+            description = c.kind,
+            origin = Option(c.origin),
+            name = c.kind
+          )
+        )
+
+      attributeSeedsCategories = attributeSeedsCategoriesKinds ++ attributeSeedsCategoriesNames
+
+      institutions <- getAllPages(50)((page, limit) =>
+        partyRegistryService.getInstitutions(Some(page), Some(limit)).map(_.items)
+      )
+      attributeSeedsInstitutions = institutions.map(i =>
+        AttributeSeed(
+          code = Option(i.originId),
+          kind = AttributeKind.CERTIFIED,
+          description = i.description,
+          origin = Option(i.origin),
+          name = i.description
+        )
+      )
+
+      _ <- addNewAttributes(attributeSeedsCategories ++ attributeSeedsInstitutions)(bearerToken)
+    } yield ()
+  }
+
+  private def addNewAttributes(
+    attributesSeeds: Seq[AttributeSeed]
+  )(bearerToken: String)(implicit contexts: Seq[(String, String)], ec: ExecutionContext): Future[Unit] = {
+
+    // calculating the delta of attributes
+    def delta(attrs: List[PersistentAttribute]): Set[AttributeSeed] =
+      attributesSeeds.foldLeft[Set[AttributeSeed]](Set.empty)((attributesDelta, seed) =>
+        attrs
+          .find(persisted => seed.origin == persisted.origin && seed.code == persisted.code)
+          .fold(attributesDelta + seed)(_ => attributesDelta)
+      )
+
+    // create all new attributes
+    for {
+      attributesfromRM <- getAll(50)(readModelService.find[PersistentAttribute]("attributes", Filters.empty(), _, _))
+      deltaAttributes = delta(attributesfromRM.toList)
+      // The client must log in case of errors
+      _ <- Future.parCollectWithLatch(100)(deltaAttributes.toList)(attributeSeed =>
```

ATM, using `parCollectWithLatch` doesn't let you log anything, it simply discards the errors:

https://github.com/pagopa/interop-commons/blob/1.0.x/utils/src/main/scala/it/pagopa/interop/commons/utils/TypeConversions.scala#L120

---

this is copied 1:1 from the current logic, I assumed that it was correct.
Should this be addressed?

---

It can be addressed now probably. It depends on what's more important, resiliency or control over the execution. Not a while ago the job was silently failing for all the attributes and it became a silent no-op. Using "non-par" methods will make the chain fail at the first error, but it will give you the chance to log.
Logging and collecting will definitively need another overload in that commons library (that can be absolutely done if it's the preferable way).

---

I agree, it is a required activity, but to me it seems out of the scope of this PR.
I'll open a separated task for this

---

Okay, I can tackle that activity in a binary compatible way in minutes, just send me the task reference

---

[here](https://pagopa.atlassian.net/browse/PIN-3607) it is ❤️